core/app/models/spree/return_authorization.rb

```diff
@@ -61,6 +62,10 @@ def can_cancel_return_items?
 
     private
 
+      def authorize_return
+        order.authorize_return! unless order.nil?
```

Comment:
You're definitely right. It would be nice to make a `ReturnAuthorizationCreator` or `CreateReturnAuthorization` class for sure, and bonus points for making it an interactor https://github.com/collectiveidea/interactor